docs: add a note for use functional component (#922)
vuejs/vue-router-next: docs/api/index.md

- **See Also**: [Meta fields](../guide/advanced/meta.md)

:::tip

> If you want to use a functional component, make sure to add a `displayName` to the component.

Reviewer: ```suggestion
If you want to use a functional component, make sure to add a `displayName` to it.
```

Author: ok

For example:
```js

> const Functional = () => h('div', 'functional')

Reviewer: Something More contextual is better
```suggestion
const HomeView = () => h('div', 'HomePage')
```

> HomeView.displayName = 'HomeView'  // add a "displayName" to the component.

Reviewer: ```suggestion
// in TypeScript, you will need to use the FunctionalComponent type
HomeView.displayName = 'HomeView'
```

Author: ok

> const router = createRouter({ path: '/foo', component: Functional })

Reviewer: can you replace this with a `const routes = [{ ... }` like in the rest of the docs?

> const router = createRouter({ path: '/foo', component: Functional })

Reviewer: Also having `/` instead of `/foo` would be better
